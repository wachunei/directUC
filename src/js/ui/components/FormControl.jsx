import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Label = styled.span`
  display: block;
  font-weight: 500;
  font-size: 14px;
`;

const Caption = styled.span`
  margin-top: 0.2rem;
  display: block;
  font-size: 12px;
`;

const ControlContainer = styled.div`
  ${Label} {
    margin-bottom: 0.2rem;
  }
  margin-bottom: 0.5rem;
`;

function FormControl({ label, caption, children }) {
  return (
    <ControlContainer>
      {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
      <label>
        {label && <Label>{label}</Label>}
        {children}
        {caption && <Caption>{caption}</Caption>}
      </label>
    </ControlContainer>
  );
}

FormControl.propTypes = {
  label: PropTypes.string,
  caption: PropTypes.string,
  children: PropTypes.node.isRequired,
};

FormControl.defaultProps = {
  label: undefined,
  caption: undefined,
};

export default FormControl;
