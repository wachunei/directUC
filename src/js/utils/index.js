export const removeSuffix = (suffix, value) =>
  value.endsWith(suffix) ? value.slice(0, -suffix.length) : value;

export const parseDistinguishedName = (name) =>
  name.replace(
    /\w\S*/g,
    (txt) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
  );

export const isFirefox = !window.chrome.app;
