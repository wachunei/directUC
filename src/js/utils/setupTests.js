// eslint-disable-next-line import/no-extraneous-dependencies
import "jest-styled-components";
