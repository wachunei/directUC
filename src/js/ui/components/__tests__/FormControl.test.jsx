import React from "react";
import FormControl from "../FormControl";
import { renderTree } from "./utils";
import "jest-styled-components";

describe("FormControl Component", () => {
  test("should match snapshot", () => {
    const tree = renderTree(<FormControl>control</FormControl>);
    expect(tree).toMatchSnapshot();
  });
  test("should render label", () => {
    const tree = renderTree(<FormControl label="A label">control</FormControl>);
    expect(tree).toMatchSnapshot();
  });
  test("should render caption", () => {
    const tree = renderTree(
      <FormControl label="A label" caption="A caption">
        control
      </FormControl>
    );
    expect(tree).toMatchSnapshot();
  });
});
