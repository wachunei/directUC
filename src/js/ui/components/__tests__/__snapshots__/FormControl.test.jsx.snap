// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormControl Component should match snapshot 1`] = `
.c0 {
  margin-bottom: 0.5rem;
}

.c0 .sc-bdnxRM {
  margin-bottom: 0.2rem;
}

<div
  class="c0"
>
  <label>
    control
  </label>
</div>
`;

exports[`FormControl Component should match snapshot with darkTheme 1`] = `
.c0 {
  margin-bottom: 0.5rem;
}

.c0 .sc-bdnxRM {
  margin-bottom: 0.2rem;
}

<div
  class="c0"
>
  <label>
    control
  </label>
</div>
`;

exports[`FormControl Component should match snapshot with lightTheme 1`] = `
.c0 {
  margin-bottom: 0.5rem;
}

.c0 .sc-bdnxRM {
  margin-bottom: 0.2rem;
}

<div
  class="c0"
>
  <label>
    control
  </label>
</div>
`;

exports[`FormControl Component should render caption 1`] = `
.c2 {
  display: block;
  font-weight: 500;
  font-size: 14px;
}

.c3 {
  margin-top: 0.2rem;
  display: block;
  font-size: 12px;
}

.c0 {
  margin-bottom: 0.5rem;
}

.c0 .c1 {
  margin-bottom: 0.2rem;
}

<div
  class="c0"
>
  <label>
    <span
      class="c1 c2"
    >
      A label
    </span>
    control
    <span
      class="c3"
    >
      A caption
    </span>
  </label>
</div>
`;

exports[`FormControl Component should render label 1`] = `
.c2 {
  display: block;
  font-weight: 500;
  font-size: 14px;
}

.c0 {
  margin-bottom: 0.5rem;
}

.c0 .c1 {
  margin-bottom: 0.2rem;
}

<div
  class="c0"
>
  <label>
    <span
      class="c1 c2"
    >
      A label
    </span>
    control
  </label>
</div>
`;
